src/app/modules/gb-analysis-module/fractalis-components/gb-fractalis-control/fractalis-chart-variable-mapper.ts

```diff
@@ -0,0 +1,150 @@
+/**
+ * Copyright 2017 - 2018  The Hyve B.V.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+import {FractalisChartDescription} from '../../../../models/fractalis-models/fractalis-chart-description';
+import {FractalisData} from '../../../../models/fractalis-models/fractalis-data';
+import {Concept} from '../../../../models/constraint-models/concept';
+import {FractalisChart} from '../../../../models/fractalis-models/fractalis-chart';
+import {ChartType} from '../../../../models/chart-models/chart-type';
+import {ConceptType} from '../../../../models/constraint-models/concept-type';
+import {FractalisChartVariable} from '../../../../models/fractalis-models/fractalis-chart-variable';
+
+
+export class ValidationError {
+  message: string;
+  constructor(message: string) {
+    this.message = message;
+  }
+}
+
+
+export class FractalisChartVariableMapper {
+
+  static parameterKeys = [
+    'groupVars',
+    'durationVars',
+    'observedVars',
+    'numVars',
+    'catVars'
+  ];
+
+  static getParameterKeys(specification: FractalisChartDescription): string[] {
+    return Object.keys(specification).filter(key => this.parameterKeys.includes(key));
+  }
+
+  /**
+   * Validates the parameter mapping with the specifies parameter constraints.
+   *
+   * @param {Map<string, string[]>} mapping
+   * @param {FractalisChartDescription} specification
+   * @return {ValidationError[]}
+   */
+  static validateMapping(mapping: Map<string, string[]>, specification: FractalisChartDescription): ValidationError[] {
+    const errors: ValidationError[] = [];
+    for (let key of this.getParameterKeys(specification)) {
+      const label = specification[key].label;
+      let variableTypeSpecification: FractalisChartVariable = specification[key];
+      let declaredVariables = mapping.get(key);
+      const minLength = variableTypeSpecification.minLength;
+      const maxLength = variableTypeSpecification.maxLength;
```

Comment:
It seems like something is missing in this validation logic. If `minLength` and `maxLength` are not specified, for me it means that variable type is not supported for the current chart. Currently this is not validated, e.g. for volcano plot or heatmap I can add categorical variables, even though the descriptions of these charts specify only the numVars.

Reply:
works fine now